code/obj/foamedmetal.dm

```diff
@@ -43,7 +43,8 @@
 	blob_act(var/power)
 		dispose()
 
-	bullet_act()
+	bullet_act(obj/projectile/P)
+		P.collide(src.loc)
 		if(metal==1 || prob(50))
 			dispose()
```

Comment:
I think that wrapping the dispose call in `SPAWN(0)` would be preferable to recursively calling `collide()` here